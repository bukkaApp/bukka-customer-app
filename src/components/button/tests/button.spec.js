import React from 'react';
import Button from '../Button';

describe('Button component', () => {
  const props = {
    text: 'SIGN IN',
    classNames: 'small-button',
    type: 'button',
    handleClick: jest.fn()
  };
  const wrapper = shallow(<Button {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props().children).toBe(props.text);
    expect(wrapper.props().className).toBe(props.classNames);
  });

  it('triggers the handleclick function when clicked', () => {
    wrapper.simulate('click');

    expect(props.handleClick).toBeCalled();
  });
});
