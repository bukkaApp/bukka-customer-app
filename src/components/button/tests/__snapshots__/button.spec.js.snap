// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders properly 1`] = `ShallowWrapper {}`;
